applications/AdjointFluidApplication/custom_schemes/adjoint_steady_velocity_pressure_scheme.h

```diff
@@ -143,42 +132,34 @@ class AdjointSteadyScheme : public Scheme<TSparseSpace, TDenseSpace>
             KRATOS_THROW_ERROR(
                 std::runtime_error, "DOMAIN_SIZE != WorkingSpaceDimension", "")
 
-        if (rModelPart.HasSubModelPart(mBoundaryModelPartName) == false)
-        {
-            KRATOS_THROW_ERROR(
-                std::runtime_error,
-                "invalid parameters \"boundary_model_part_name\": ",
-                mBoundaryModelPartName)
-        }
-
-        mpObjectiveFunction->Initialize(rModelPart);
+        mpResponseFunction->Initialize();
 
         KRATOS_CATCH("")
     }
 
     void InitializeSolutionStep(ModelPart& rModelPart,
-                                        SystemMatrixType& rA,
-                                        SystemVectorType& rDx,
-                                        SystemVectorType& rb) override
+                                SystemMatrixType& rA,
+                                SystemVectorType& rDx,
+                                SystemVectorType& rb) override
     {
         KRATOS_TRY
 
+        // Sensitivities are generally computed as a time integral. For steady
+        // problems, we set the time step to -1.0 (minus because adjoint is
+        // backward in time).
+        rModelPart.GetProcessInfo()[DELTA_TIME] = -1.0;
+
         BaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
 
         // initialize the variables to zero.
         for (auto it = rModelPart.NodesBegin(); it != rModelPart.NodesEnd(); ++it)
```

Comment:
The reason for multiple ids is due to the threads. It checks if the most recently calculated vector in the current thread was for the current element. If true, it skips the loop over all the nodes and just reuses the vector which is stored. I added this because for transient adjoint, the same vector is accessed multiple times for a single element in a thread.